src/utils/hooks/useLoadApplicationNEW.ts

```diff
@@ -98,11 +136,11 @@ const useLoadApplication = ({ serialNumber, networkFetch }: UseGetApplicationPro
       sections: buildSectionsStructure({ sections, baseElements }),
     })
     setIsLoading(false)
-  }, [data])
+  }, [data, loading])
```

Comment:
Is it totally necessary to monitor `loading` here? Isn't a change in `data` enough?  (If so, I'll need to change it on the other hook too)

Reply:
As per comment in telegram:

should monitor loading, error and data.

in use effect first check loading
then error
then !data (just in case) <- technically shouldn't need to

Reply:
That is required to run the useEffect again ofter the second try of refetch - the `data` doesn't change because the result was the same, but we still need to set a second refetch to try and get the correct status of the application. So the way to force the useEffect on running again was to watch the loading property - and only continue if `loading` === `false`